react/features/base/media/actions.js

```diff
@@ -92,6 +94,47 @@ export function setCameraFacingMode(cameraFacingMode: string) {
     };
 }
 
+/**
+ * Action to set the muted state of the local screenshare.
+ *
+ * @param {boolean} muted - True if the local screenshare is to be enabled or false otherwise.
+ * @param {MEDIA_TYPE} mediaType - The type of media.
+ * @param {number} authority - The {@link SCREENSHARE_MUTISM_AUTHORITY} which is muting/unmuting the local screenshare.
+ * @param {boolean} ensureTrack - True if we want to ensure that a new track is created if missing.
+ * @returns {Function}
+ */
+export function setScreenshareMuted(
+        muted: boolean,
+        mediaType: MediaType = MEDIA_TYPE.SCREENSHARE,
+        authority: number = SCREENSHARE_MUTISM_AUTHORITY.USER,
+        ensureTrack: boolean = false) {
+    return (dispatch: Dispatch<any>, getState: Function) => {
+        const state = getState();
+
+        // check for A/V Moderation when trying to unmute
+        if (!muted && shouldShowModeratedNotification(MEDIA_TYPE.SCREENSHARE, state)) {
+            if (!isModerationNotificationDisplayed(MEDIA_TYPE.SCREENSHARE, state)) {
+                ensureTrack && dispatch(showModeratedNotification(MEDIA_TYPE.SCREENSHARE));
+            }
+
+            return;
+        }
+
+        const oldValue = state['features/base/media'].screenshare.muted;
+
+        // eslint-disable-next-line no-bitwise
+        const newValue = muted ? oldValue | authority : oldValue & ~authority;
```

Comment:
it's hard to understand this logic, but I guess it's copy pasted from the existing camera mute handling?

Reply:
Yes, wanted to keep the muted by audio-only behavior same across camera/desktop tracks.